public/coffee/ide/pdfng/directives/pdfRenderer.coffee

```diff
@@ -118,17 +129,38 @@ define [
 
 				completeRef = @complete
 				renderTaskRef = @renderTask
+				# console.log 'started page load', pagenum
+
+				timedOut = false
+				timer = $timeout () =>
+					Raven.captureMessage?('pdfng page load timed out after ' + @PAGE_LOAD_TIMEOUT + 'ms')
+					# console.log 'page load timed out', pagenum
+					timedOut = true
+					@spinner.stop(element.canvas)
+					# @jobs = @jobs - 1
+					# @triggerRenderQueue(0)
+					this.errorCallback?('timeout')
+				, @PAGE_LOAD_TIMEOUT
```

Comment:
On successful page load the timeout is immediately cancelled before any rendering is started so this should be ok as only one page is rendered at any time.